editor/test/integration/model_test.clj

```diff
@@ -33,16 +34,36 @@
 
 (deftest textures
   (test-util/with-loaded-project
-    (let [node-id (test-util/resource-node project "/model/test.model")]
-      (let [original-texture (first (test-util/prop node-id :textures))
-            t [original-texture nil nil]]
-        (test-util/prop! node-id :textures t)
-        (is (= t (test-util/prop node-id :textures)))
-        (let [p (-> [(g/node-value node-id :_properties)]
-                  (properties/coalesce)
+    (let [node-id (test-util/resource-node project "/model/test.model")
+          original-materials (test-util/prop node-id :materials)
+          original-texture (get-in original-materials [0 :textures 0 :texture])]
+      (test-util/prop! node-id :materials (assoc-in original-materials [0 :textures 0 :texture] original-texture))
+      (is (= original-materials (test-util/prop node-id :materials)))
+      (let [p (-> (properties/coalesce [(g/node-value node-id :_properties)])
                   :properties
-                  :texture2)]
-          (properties/set-values! p [original-texture]))))))
+                  :__sampler__tex2__default)]
+        (properties/set-values! p [original-texture]))))
+  (testing "Loading textures renames them to match the material"
+    (test-util/with-loaded-project "test/resources/model_migration_project"
+      ;; test2.model has textures that don't match the names of the material, so
+      ;; loading the model renames the texture bindings
+      (let [node-id (test-util/resource-node project "/test.model")]
+        (is (g/node-value node-id :dirty?))
+        (is (= #{"tex_foo" "tex_bar" "tex_baz"}
+               (->> (g/node-value node-id :resource)
+                    slurp
+                    (protobuf/str->map ModelProto$ModelDesc)
+                    :materials
+                    (eduction
+                      (mapcat :textures)
+                      (map :sampler))
+                    (into #{}))))
+        (is (= #{"tex0" "tex1" "tex2"}
+               (->> (g/node-value node-id :materials)
+                    (eduction
+                      (mapcat :textures)
+                      (map :sampler))
+                    (into #{}))))))))
```

Comment:
I think using `->>` and `eduction` here becomes more lines than the respective `(into ...)` expressions would have been?

Reply:
I argue not using `->>` at all is simpler still:
```clj
(into #{}
      (comp (mapcat :textures)
            (map :sampler))
      (g/node-value node-id :materials))
```
But maybe not in the first case.

You could combine `slurp` and `str->map` into `protobuf/read-text` if you wanted, though!